File: LeetCodeProblems/easyProblems/questionFour/solutionFour.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int lenWrdOne = word1.length();
        int lenWrdTwo = word2.length();
        int loopOverThis = lenWrdOne + lenWrdTwo;
        int counter = 0;
        String merged = "";

        while (counter < loopOverThis) {
            if (counter < lenWrdOne) {
                try {
                    merged = merged + word1.charAt(counter);
                } catch (Exception e) {
                    // Do nothing
                }
            }
            if (counter < lenWrdTwo) {
                try {
                     merged = merged + word2.charAt(counter);
                } catch (Exception e) {
                    // Do nothing
                }
            }
            counter++;
        }

        return merged;
    }
}

File: LeetCodeProblems/easyProblems/questionFour/solutionfour.cpp
#include <iostream>
#include <string>


class Solution {
public:
    std::string mergeAlternately(std::string word1, std::string word2) {
        int lengthWordOne = word1.length();
        int lengthWordTwo = word2.length();
        int loopOverThis = lengthWordOne+lengthWordTwo;
        std::string  mergedString;
        int counter = 0;

        while(counter < loopOverThis){
            if (counter < lengthWordOne){
                try{
                    mergedString += word1[counter];
                }catch (...){
                    // Do nothing
                }
            }
            if (counter < lengthWordTwo){
                try{
                    mergedString += word2[counter];
                }catch (...){
                    // Do nothing
                }
            }
             counter++;
        }

        return mergedString;
    }

};

int main(){

    Solution sol;
    std::string result;
    result= sol.mergeAlternately("Anam","Wahid");
    std::cout<<result;

    return 0;
}